### ISSUE TITLE
Stalled/Pending/Failed not being recorded in the data files

### ISSUE THREAD

**Issue by adam704a** [instedd/ask#762]
Steps
1. Start a survey with an invalid number
2. Observe that the invalid numbers show up as failed in the overview (http://ask.instedd.org/projects/105/surveys/374)
3. Click on the respondent's tab

Expected: 
The failed cases would appear as such in the respondent's tab the failed cases could be download as failed.

Actual
The disposition codes are not present for failed cases. The failed cases are not indicated as such in the survey response file/disposition history file.

* Same for stalled cases.

[failed-disposition.zip](https://github.com/instedd/ask/files/856286/failed-disposition.zip)

@steesdale

**Comment by spalladino** [comment]
Show "failed" as disposition in all downloads, for respondents in failed state that have no internal disposition set. When marking the respondent as _failed_, set disposition to _failed_ as well if the respondent had not been flagged as partial/ineligible/complete.

**Comment by spalladino** [comment]
@adam704a how would you expect Stalled cases to be presented? Note that stalled respondents are automatically moved to failed when the survey ends, so you would see a transition from stalled to failed for all respondents once the survey is finished. Also, a respondent can "leave" the stalled state if he/she replies.

We'd advise *not* to consider stalled as a disposition, but could you tell us what information you are trying to gather from this state?

**Comment by adam704a** [comment]
We would like to have Stalled and Pending dispositions available in the download, to match the overview screen. Having this disposition will allow administrators to run analysis on those users who have stopped responding (stalled) and those who haven't been scheduled (pending). The biggest thing here that if the disposition is listed on the overview screen, it should be in the download file.

**Comment by adam704a** [comment]
@spalladino @steesdale: this is **not** needed for Gingko 2 (0.7.2)

**Comment by nditada** [comment]
![ncd - disposition diagram](https://cloud.githubusercontent.com/assets/1359189/24430405/cbeb38ae-13ec-11e7-9d8e-429d7533e9a1.png)
@adam704a I'm attaching the state diagrama as support for the ongoing conversation we started yesterday on this. As I mentioned, there are some "states" that are not being translated to "dispositions", mostly because they imply a temporary state while the survey is running. But it could be changed. Let's discuss what are the goals.

**Comment by nekron** [comment]
This should've been fixed in 0.9 (Ironwood)

**Comment by pmallol** [comment]
As for Version: 0.9.0 (build 2357), respondents that appear as 'Failed' in Survey Overview due to the following error in verboice:
 
```State: Failed
Failure information: Unknown error. Interworking, unspecified (127)
```
show up in CSVs as it follows:
- Disposition 'Failed' in Survey Results CSV
- 'Queued' in Disposition History CSV
- Disposition 'Queued' in Interactions CSV

- Incentives CSV is empty
- Disposition column in Respondents tab is also empty

![screen shot 2017-05-24 at 2 24 23 pm](https://cloud.githubusercontent.com/assets/9214460/26416570/b88011f0-408c-11e7-8a5a-e51c7de8071f.png)
![screen shot 2017-05-24 at 2 25 06 pm](https://cloud.githubusercontent.com/assets/9214460/26416596/d5aab00a-408c-11e7-8c2f-44dbc641abb6.png)
![screen shot 2017-05-24 at 2 25 31 pm](https://cloud.githubusercontent.com/assets/9214460/26416612/e403bf0c-408c-11e7-8832-f4a1a9a2c38c.png)

[762-csv.zip](https://github.com/instedd/ask/files/1026432/762-csv.zip)

Expected:
- Disposition column in Respondents tab should display 'Failed'
- No CSV should show 'Queued' as a Disposition
